src/main/java/com/uid2/admin/vertx/service/SiteService.java

```diff
@@ -330,6 +348,38 @@ private void handleSiteDomains(RoutingContext rc) {
         }
     }
 
+    private void handleSiteAppNames(RoutingContext rc) {
+        try {
+            // refresh manually
+            siteProvider.loadContent();
+
+            final Site existingSite = RequestUtil.getSiteFromParam(rc, "id", siteProvider);
+            if (existingSite == null) {
+                return;
+            }
+
+            JsonObject body = rc.body().asJsonObject();
+            JsonArray appNamesJa = body.getJsonArray("app_names");
+            if (appNamesJa == null) {
+                ResponseUtil.error(rc, 400, "required parameters: app_names");
+                return;
+            }
+            List<String> normalizedAppNames = getNormalizedAppNames(rc, appNamesJa);
+            if (normalizedAppNames == null) return;
+
+            existingSite.setAppNames(new HashSet<>(normalizedAppNames));
```

Comment:
since `setAppNames()` accepts a Set perhaps we can return a Set from `getNormalizedAppNames()` as well ?